hathor/transaction/base_transaction.py

```diff
@@ -519,6 +519,16 @@ def validate_full(self, skip_block_weight_verification: bool = False, sync_check
         from hathor.transaction.transaction_metadata import ValidationState
 
         meta = self.get_metadata()
+
+        # XXX: blocks can have been created previously with soft_height, we'll remove it and add a proper height
+        if self.is_block:
+            soft_height = meta.soft_height
+            meta.soft_height = None
+            meta.height = self.calculate_height()
+            if soft_height is not None and soft_height != meta.height:
+                self.log.error('A wrong height was previously set as soft height, this indicates a but',
```

Comment:
Shouldn't we stop the full node in this case? Or ban the peer that sent us this stream of blocks?

Reply:
I'm not sure we have a good precedent for "stopping" the full node. I prefer leaving it as an assert, which I'd say is the next hard error in line (after a stop).

Reply:
Outdated conversation.